File: backend/app/models/organization.py
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, ForeignKey, func, Text
from sqlalchemy.orm import relationship
from app.database.database import Base


class Organization(Base):
    __tablename__ = "organizations"

    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("scraping_tasks.id"), nullable=False)

    name = Column(String(500), nullable=False)
    category = Column(String(255), nullable=True)
    location = Column(String(255), nullable=True)

    # Address fields
    address = Column(Text, nullable=True)
    street = Column(String(500), nullable=True)
    area = Column(String(255), nullable=True)
    city = Column(String(255), nullable=True)
    state = Column(String(255), nullable=True)
    pincode = Column(String(10), nullable=True)
    address_source = Column(Text, nullable=True)

    # Confidence
    confidence_score = Column(Float, default=0.0)
    confidence_level = Column(String(10), default="LOW")  # LOW/MEDIUM/HIGH

    # Meta
    source = Column(String(255), nullable=True)
    is_duplicate = Column(Boolean, default=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    task = relationship("ScrapingTask", back_populates="organizations")
    websites = relationship("Website", back_populates="organization", cascade="all, delete-orphan")
    contacts = relationship("Contact", back_populates="organization", cascade="all, delete-orphan")
    phone_numbers = relationship("PhoneNumber", back_populates="organization", cascade="all, delete-orphan")
    email_addresses = relationship("EmailAddress", back_populates="organization", cascade="all, delete-orphan")
    social_links = relationship("SocialLink", back_populates="organization", cascade="all, delete-orphan")
    source_pages = relationship("SourcePage", back_populates="organization", cascade="all, delete-orphan")
    saved_by = relationship("SavedLead", back_populates="organization", cascade="all, delete-orphan")

File: backend/src/models/Organization.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Organization = sequelize.define('Organization', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  task_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  name: {
    type: DataTypes.STRING(500),
    allowNull: false,
  },
  category: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  location: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  address: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  street: {
    type: DataTypes.STRING(500),
    allowNull: true,
  },
  area: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  city: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  state: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  pincode: {
    type: DataTypes.STRING(10),
    allowNull: true,
  },
  address_source: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  confidence_score: {
    type: DataTypes.FLOAT,
    defaultValue: 0.0,
  },
  confidence_level: {
    type: DataTypes.STRING(10),
    defaultValue: 'LOW',
  },
  source: {
    type: DataTypes.STRING(255),
    allowNull: true,
  },
  is_duplicate: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
  updated_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'organizations',
  timestamps: false,
});

module.exports = Organization;
